sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -534,13 +521,16 @@ def _open(self):
                 self._handler.close()
             auth = self._client._create_auth()
             self._create_handler(auth)
-            self._handler.open()
+            conn = self._client._conn_manager.get_connection(  # pylint: disable=protected-access
+                host=self._client._address.hostname, auth=auth
+            )
+            self._handler.open(connection=conn)
             while not self._handler.client_ready():
                 time.sleep(0.05)
             self._max_message_size_on_link = (
-                self._handler._link.remote_max_message_size
-                or constants.MAX_FRAME_SIZE_BYTES
```

Comment:
created an issue here to add a MAX_MESSAGE_LENGTH_BYTES constant to pyamqp: https://github.com/Azure/azure-sdk-for-python/issues/25875

For now, MAX_MESSAGE_LENGTH_BYTES = MAX_FRAME_SIZE_BYTES for pyamqp